cmd/pcr0tool/commands/diff/command.go

```diff
@@ -225,30 +225,48 @@ func (cmd Command) Execute(args []string) {
 	switch outputFormat {
 	case outputFormatTypeAnalyzedText:
 		output, err := format.AsText(
-			diff.Analyze(diffEntries, measurements, firmwareGood, firmwareBadData),
-			debugInfo, measurements, firmwareGoodData, firmwareBadData,
+			diff.Analyze(diffEntries, measurementsForDiffAnalysis(measurements), firmwareGood, firmwareBadData),
+			debugInfo, firmwareGoodData, firmwareBadData,
 		)
 		assertNoError(err)
 		fmt.Print(output)
 	case outputFormatTypeAnalyzedJSON:
 		outputAnalyzedJSON(
-			diff.Analyze(diffEntries, measurements, firmwareGood, firmwareBadData),
+			diff.Analyze(diffEntries, measurementsForDiffAnalysis(measurements), firmwareGood, firmwareBadData),
 			debugInfo, measurements,
 		)
 	case outputFormatTypeJSON:
 		outputJSON(diffEntries, debugInfo, measurements)
 	}
 }
 
-// MeasurementsLaconic is a helper to print measurements in a laconic way
-type MeasurementsLaconic pcr.Measurements
+func measurementsForDiffAnalysis(ms pcr.Measurements) diff.Measurements {
+	result := make(diff.Measurements, 0, len(ms))
+	for _, m := range ms {
+		result = append(result, measurementForDiffAnalysis(m))
+	}
+	return result
+}
+
+func measurementForDiffAnalysis(m *pcr.Measurement) diff.Measurement {
+	result := diff.Measurement{
+		Description: m.ID.String(),
+		Chunks:      make(diff.DataChunks, 0, len(m.Data)),
+		CustomData:  m,
```

Comment:
Package `pkg/diff` gets two firmware images and the diverged ranges coordinates as the input, and provides explanation of what is going on in these diverged ranges. This was initially written to explain reports generated by `pcr`, but now it is also useful to explain reports generated by `bootflow`. So here I move to neutral types, which does not have a dependency on `pcr` (so that I can reuse the same code in `bootflow`).

Let me know if I misunderstood the question and haven't answered it :)

Reply:
`CustomData` here is of type `*pcr.Measurement` and I don't see how it can be usefully used within the diff package, nor where it's actually used at all. Your comments make sense in the context of the rest of the structure elements, it's just this embedding of the thing we're trying to decouple that I don't follow.